pkg/policy/manager.go

```diff
@@ -87,49 +88,54 @@ func NewManager(policyDir string, pool *pgxpool.Pool, logger hclog.Logger) Manag
 	}
 }
 
-// ParsePolicyHubPath parses and validates the given arguments into the Policy struct.
-// Given args should follow the following semantic structure:
-// [(organization/)repository-name(@tag)] ([repository-path])
-func (m *ManagerImpl) ParsePolicyHubPath(args []string, subPolicyPath string) (*Policy, error) {
-	// Make sure the mandatory args are given
-	if len(args) < 1 {
-		return nil, fmt.Errorf("invalid policy path. Repository name is required but got %#v", args)
-	}
-	policy := &Policy{
-		SubPath: subPolicyPath,
+func (m *ManagerImpl) Run(ctx context.Context, execReq *ExecuteRequest, policyWrapper *PolicyWrapper) (*ExecutionResult, error) {
+	// Acquire connection from the connection pool
+	conn, err := m.pool.Acquire(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to acquire connection from the connection pool: %s", err.Error())
 	}
+	defer conn.Release()
 
-	// Parse and validate org/repository
-	orgRepoSplit := strings.Split(args[0], pathDelimiter)
-
-	// Parse org/repo
-	switch len(orgRepoSplit) {
-	case 2:
-		policy.Organization = orgRepoSplit[0]
-		policy.Repository = orgRepoSplit[1]
-	case 1:
-		policy.Repository = orgRepoSplit[0]
-		policy.Organization = cloudQueryOrg
-	default:
-		return nil, fmt.Errorf("invalid policy path. Repository name malformed: %s", args[0])
+	var totalQueriesToRun = policyWrapper.getQueriesCount()
+	var finishedQueries = 0
+
+	// set the progress total queries to run
+	if execReq.UpdateCallback != nil {
+		execReq.UpdateCallback(Update{
+			PolicyName:      execReq.Policy.Name,
+			Version:         execReq.Policy.Version,
+			FinishedQueries: 0,
+			QueriesCount:    totalQueriesToRun,
+			Error:           "",
+		})
 	}
 
-	// Parse version
-	versionSplit := strings.Split(policy.Repository, versionDelimiter)
-	if len(versionSplit) == 2 {
-		policy.Version = versionSplit[1]
-		policy.Repository = versionSplit[0]
+	// replace console update function to keep track the current status
+	var progressUpdate = func(update Update) {
+		finishedQueries += update.FinishedQueries
+		if execReq.UpdateCallback != nil {
+			execReq.UpdateCallback(Update{
+				PolicyName:      execReq.Policy.Name,
+				Version:         execReq.Policy.Version,
+				FinishedQueries: finishedQueries,
+				QueriesCount:    totalQueriesToRun,
+				Error:           "",
+			})
+		}
 	}
 
-	// Parse repository path if given
-	if len(args) == 2 {
-		policy.RepositoryPath = args[1]
+	executor := NewExecutor(conn, m.logger)
```

Comment:
I think we can pass the updatecallback, and the execute as well perhaps when initializing the Executor, seems like it will be reduce the amount of arguments passed